### PR TITLE
Bump LLVM and GPUCompiler versions + related fixes

### PR DESCRIPTION
**Pull request by gbaraldi**
The fixes entail
Drop abi_call for a plain invoke call
Change to use string api for pass manager
Drop support for versions before 1.10

~1.11 still segfaults so I'm taking a further look~
1.11 passes tests

### CODE REVIEW

File: src/AllocCheck.jl [hunk 1]
```diff
@@ -93,12 +93,28 @@ Find all static allocation sites in the provided LLVM IR.
 
 This function modifies the LLVM module in-place, effectively trashing it.
 """
-function find_allocs!(mod::LLVM.Module, meta; ignore_throw=true)
+function find_allocs!(mod::LLVM.Module, meta, entry_name::String; ignore_throw=true, invoke_entry=false)
```
Comment (topolarity) [inline src/AllocCheck.jl:96]: Can you document what `invoke_entry` does?

**Comment by topolarity** [comment]
Looking pretty good, but I'd like to ideally get nightly working:
```julia
[91723] signal 11 (1): Segmentation fault
in expression starting at /home/topolarity/repos/AllocCheck/test/runtests.jl:181
_ZNK4llvm11LLVMContext14getOptPassGateEv at /home/topolarity/repos/julia/usr/bin/../lib/libLLVM.so.18.1jl (unknown line)
unknown function (ip: (nil))
Allocations: 24540140 (Pool: 24539670; Big: 470); GC: 18
ERROR: Package AllocCheck errored during testing (received signal: 11)
```

**Comment by ExpandingMan** [comment]
This bound has been holding a lot back for me.  It's been two weeks, any chance you can merge and deal with nightly later?

**Comment by topolarity** [comment]
I was able to most of nightly working, thankfully 👍 

Bug for the skipped test is in https://github.com/JuliaLang/AllocCheck.jl/issues/84

**Review by topolarity — APPROVED** [review]
Thanks @gbaraldi  - We'll have to follow-up to add the docs and fix the lingering segfault, but let's get this in for now.

**Pull request merged**